doc/cookbook/canary-release.md

```diff
@@ -0,0 +1,280 @@
+
+We provided easy way to parse user-agent in [easegress-rust-uaparser](https://github.com/megaease/easegress-rust-uaparser). Download `easegress-rust-uaparser`
+
+```bash
+git clone https://github.com/megaease/easegress-rust-uaparser.git
+```
+
+Enter the `easegress-rust-uaparser/cloudflare` directory and deploy the worker using the wrangler command
+
+```bash
+npx wrangler publish
+```
+
+Enter the Cloudflare website configuration page and configure Workers Routes as you needed.
+
+2. Traffic scheduling rules:
+
+Save the below YAML to ecommerce-service.yaml, and make sure you have replaced the servers with yours.
+
+```yaml
+name: ecommerce-pipeline
+kind: Pipeline
+flow:
+- filter: buildOrderRequest
+  namespace: order
+- filter: order-service
+  namespace: order
+- filter: buildNotifyRequest
+  namespace: notify
+- filter: notify-service
+  namespace: notify
+- filter: buildResponse
+filters:
+- name: buildOrderRequest
+  kind: RequestBuilder
+  template: |
+    url: /order
+- name: order-service
+  kind: Proxy
+  pools:
+  - servers:
+    - url: http://megaease.com/api/v2
+- name: buildNotifyRequest
+  kind: RequestBuilder  # use order response as notify request body
+  template: |
+    url: /notify
+    method: POST
+    body: |
+      {"order": "{{.responses.order.Body | jsonEscape}}"}
+- name: notify-service
+  kind: Proxy
+  pools:
+  - servers:
+    - url: http://megaease.com/api/v1
+  - filter:
+      headers:
+        x-ua-device: # Cloudflare's user device header
+          exact: Mac # the traffic from Mac should be directed to the new version
+    servers:
+    - url: http://megaease.com/api/v2
+- name: buildResponse
+  kind: ResponseBuilder
+  template: |
+    statusCode: {{.responses.notify.StatusCode}}
+    body: |
+      [{{.responses.notify.Body}}]
+```
+
+Then update the E-commerce pipeline with the command:
+
+```bash
+egctl object update -f ecommerce-service.yaml
+```
+
+## based on user OS
+
+![case 3](../imgs/canary-release-case-3.png)
+
+Suppose order service and notify service have upgraded at the same time. The user who is accessing our service from a Mac device should be directed to the new version of the order service (v2), and from MacOS should be directed to the new version of the notify service (v2). 
+
+You may think Cloudflare's Worker is not suitable for you, there is another easy way to parse user-agent with Easegress's WASM.
+
+1. Traffic tagging: using Easegress's WASM to parse user-agent. 
+
+Download `easegress-rust-uaparser`
+
+```bash
+git clone https://github.com/megaease/easegress-rust-uaparser.git
+```
+
+Navigate to the `easegress-rust-uaparser/binary` directory and use the `easegress.wasm` file directly. It will parse user-agent and add the `x-ua-device` and `x-ua-os` headers to the HTTP request.
```

Comment:
```suggestion
Go to the `easegress-rust-uaparser/binary` directory and just use the `easegress.wasm` file directly. It will parse the `User-Agent` and add the `x-ua-device` and `x-ua-os` headers to the HTTP request.
```